src/views/Pools/components/CakeVaultCard/RecentCakeProfitBalance.tsx

```diff
@@ -2,20 +2,18 @@ import React from 'react'
 import BigNumber from 'bignumber.js'
 import { TooltipText, useTooltip } from '@pancakeswap-libs/uikit'
 import { useTranslation } from 'contexts/Localization'
+import { useCakeVault } from 'state/hooks'
 import { getFullDisplayBalance } from 'utils/formatBalance'
 import { convertSharesToCake } from '../../helpers'
 
 interface RecentCakeProfitBalanceProps {
   cakeAtLastUserAction: BigNumber
   userShares: BigNumber
-  pricePerFullShare: BigNumber
 }
 
-const RecentCakeProfitBalance: React.FC<RecentCakeProfitBalanceProps> = ({
-  cakeAtLastUserAction,
-  userShares,
-  pricePerFullShare,
-}) => {
+const RecentCakeProfitBalance: React.FC<RecentCakeProfitBalanceProps> = ({ cakeAtLastUserAction, userShares }) => {
+  const { pricePerFullShare: pricePerFullShareAsString } = useCakeVault()
+  const pricePerFullShare = new BigNumber(pricePerFullShareAsString)
```

Comment:
Not sure if there's a more elegant solution for this?

1. Serialise BigNumber as string when storing it in Redux
2. Fetch string from Redux store
3. Initialise as BigNumber when used in code

Reply:
So I've changed this in this commit - https://github.com/pancakeswap/pancake-frontend/pull/1113/commits/92f04d72bf2037e8a7a151a70199b1f792c6ef36 

There's now an intermediary hook `useGetVaultPublicData` that parses the serialised string and returns an actual BigNumber. This feels slightly like overkill, but it _does_ save on a lot of repeated code when instantiating these strings as BigNumbers.

New hook - https://github.com/pancakeswap/pancake-frontend/pull/1113/commits/92f04d72bf2037e8a7a151a70199b1f792c6ef36#diff-6a15e2082467c89c26382ca7c5b411677f8abc61512c4ecefe4515fa5aeb4123R1-R18

Example of how this saves on doing this repeatedly in the codebase - https://github.com/pancakeswap/pancake-frontend/pull/1113/commits/92f04d72bf2037e8a7a151a70199b1f792c6ef36#diff-2fadc663a1c0af7a175e5fe2156f5eac5c330a048c2daef4586416b4dfd0a6e3L15-R15

Let me know if you think there's a better pattern for this?

Reply:
I think this is fine for now.
Other way would be to just convert all values that should be BigNumber to BigNumber inside `useCakeVault` and keep other values as strings if needed (like fees here)? Then you have just one hook for everything, destructure what you need and here you go.

Actually interesting topic of storing class instances inside global state. As I understand with redux its not possible, I wonder if Context could be better option in this case (not sure though, might have problems with that too).